src/components/ui/QuestionBoard.js

```diff
@@ -139,18 +140,26 @@ class QuestionBoard extends React.Component {
         }
     }
 
-    onButtonClick = (index, id) => {
+    onButtonClick = (index, id, key) => {
         this.state.answers[0].push(id);
         this.state.answers[1].push(index);
 
         if (this.state.count < this.props.questions.length - 1) {
-            this.setState({ count: this.state.count + 1 });
+            this.setState({ count: this.state.count + 1, prevButtonKey: key });
         } else {
             this.context.toggleRedirect();
             this.setRedirect();
         }
     }
 
+    getButtonKey = (index) => {
```

Comment:
Can it be tied up? like if() else if() else() and this.state.prevButtonKey s on the same side.

Reply:
Done! getButtonKey was removed